flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/files/StaticFileServerHandler.java

```diff
@@ -205,215 +125,18 @@ public void channelRead0(ChannelHandlerContext ctx, Routed routed) throws Except
-	 * @param status The response status.
-	 */
-	private static void sendError(ChannelHandlerContext ctx, HttpResponseStatus status) {
-		FullHttpResponse response = new DefaultFullHttpResponse(
-				HTTP_1_1, status, Unpooled.copiedBuffer("Failure: " + status + "\r\n", CharsetUtil.UTF_8));
-		response.headers().set(CONTENT_TYPE, "text/plain; charset=UTF-8");
-
-		// close the connection as soon as the error message is sent.
-		ctx.writeAndFlush(response).addListener(ChannelFutureListener.CLOSE);
-	}
-
-	/**
-	 * Send the "304 Not Modified" response. This response can be used when the
-	 * file timestamp is the same as what the browser is sending up.
-	 *
-	 * @param ctx The channel context to write the response to.
-	 */
-	private static void sendNotModified(ChannelHandlerContext ctx) {
-		FullHttpResponse response = new DefaultFullHttpResponse(HTTP_1_1, NOT_MODIFIED);
-		setDateHeader(response);
-
-		// close the connection as soon as the error message is sent.
-		ctx.writeAndFlush(response).addListener(ChannelFutureListener.CLOSE);
-	}
-
-	/**
-	 * Sets the "date" header for the HTTP response.
-	 *
-	 * @param response HTTP response
-	 */
-	private static void setDateHeader(FullHttpResponse response) {
-		SimpleDateFormat dateFormatter = new SimpleDateFormat(HTTP_DATE_FORMAT, Locale.US);
-		dateFormatter.setTimeZone(GMT_TIMEZONE);
-
-		Calendar time = new GregorianCalendar();
-		response.headers().set(DATE, dateFormatter.format(time.getTime()));
-	}
-
-	/**
-	 * Sets the "date" and "cache" headers for the HTTP Response.
-	 *
-	 * @param response    The HTTP response object.
-	 * @param fileToCache File to extract the modification timestamp from.
-	 */
-	private static void setDateAndCacheHeaders(HttpResponse response, File fileToCache) {
-		SimpleDateFormat dateFormatter = new SimpleDateFormat(HTTP_DATE_FORMAT, Locale.US);
-		dateFormatter.setTimeZone(GMT_TIMEZONE);
-
-		// date header
-		Calendar time = new GregorianCalendar();
-		response.headers().set(DATE, dateFormatter.format(time.getTime()));
-
-		// cache headers
-		time.add(Calendar.SECOND, HTTP_CACHE_SECONDS);
-		response.headers().set(EXPIRES, dateFormatter.format(time.getTime()));
-		response.headers().set(CACHE_CONTROL, "private, max-age=" + HTTP_CACHE_SECONDS);
-		response.headers().set(LAST_MODIFIED, dateFormatter.format(new Date(fileToCache.lastModified())));
-	}
-
-	/**
-	 * Sets the content type header for the HTTP Response.
-	 *
-	 * @param response HTTP response
-	 * @param file     file to extract content type
-	 */
-	private static void setContentTypeHeader(HttpResponse response, File file) {
-		String mimeType = MimeTypes.getMimeTypeForFileName(file.getName());
-		String mimeFinal = mimeType != null ? mimeType : MimeTypes.getDefaultMimeType();
-		response.headers().set(CONTENT_TYPE, mimeFinal);
+	protected boolean shouldCache(String requestPath) {
+		return !(requestPath.contains("log") || requestPath.contains("out"));
```

Comment:
We could use endsWith instead. I coud also whip up some regex to detect the actual JM/TM log path.